src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```
@@ -0,0 +1,40 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new AuthenticationException("Can't authenticate user " + email));
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can't authenticate user: "
+                + "invalid email or password" + email);
```

This block contains inverted logic. If the passwords do not match, you return the user, but you should be throwing an `AuthenticationException`. Conversely, if the passwords do match, you throw an exception instead of returning the user.

This also violates checklist item #7: 'Create only one condition for throwing AuthenticationException in login() method'. The user existence check and password validation should be combined to have a single failure point that throws the exception.